workflows-scripts/users/users-loan-renewal/users_loanRenewals.jmx

```diff
@@ -0,0 +1,471 @@
+            </JSONPostProcessor>
+            <hashTree/>
+            <JSONPostProcessor guiclass="JSONPostProcessorGui" testclass="JSONPostProcessor" testname="Extract totalRecords" enabled="true">
+              <stringProp name="JSONPostProcessor.referenceNames">totalRecords</stringProp>
+              <stringProp name="JSONPostProcessor.jsonPathExprs">$.totalRecords</stringProp>
+              <stringProp name="JSONPostProcessor.match_numbers">-1</stringProp>
+              <stringProp name="JSONPostProcessor.defaultValues">NOT_FOUND;</stringProp>
+              <boolProp name="JSONPostProcessor.compute_concat">true</boolProp>
+            </JSONPostProcessor>
+            <hashTree/>
+            <ResponseAssertion guiclass="AssertionGui" testclass="ResponseAssertion" testname="Response Code Assertion" enabled="true">
+              <collectionProp name="Asserion.test_strings">
+                <stringProp name="49586">200</stringProp>
+              </collectionProp>
+              <stringProp name="Assertion.test_field">Assertion.response_code</stringProp>
+              <boolProp name="Assertion.assume_success">false</boolProp>
+              <intProp name="Assertion.test_type">8</intProp>
+              <stringProp name="Assertion.custom_message"></stringProp>
+            </ResponseAssertion>
+            <hashTree/>
+            <ResponseAssertion guiclass="AssertionGui" testclass="ResponseAssertion" testname="Response Body Assertion" enabled="true">
+              <collectionProp name="Asserion.test_strings">
+                <stringProp name="-987155039">&quot;loans&quot;</stringProp>
+              </collectionProp>
+              <stringProp name="Assertion.custom_message"></stringProp>
+              <stringProp name="Assertion.test_field">Assertion.response_data</stringProp>
+              <boolProp name="Assertion.assume_success">false</boolProp>
+              <intProp name="Assertion.test_type">2</intProp>
+            </ResponseAssertion>
+            <hashTree/>
+          </hashTree>
+          <JSR223Sampler guiclass="TestBeanGUI" testclass="JSR223Sampler" testname="InitCounter" enabled="true">
+            <stringProp name="cacheKey">true</stringProp>
+            <stringProp name="filename"></stringProp>
+            <stringProp name="parameters"></stringProp>
+            <stringProp name="script">vars.put(&quot;counter&quot;,&quot;1&quot;);</stringProp>
+            <stringProp name="scriptLanguage">java</stringProp>
+          </JSR223Sampler>
+          <hashTree/>
+          <TransactionController guiclass="TransactionControllerGui" testclass="TransactionController" testname="Loan renewal Transaction Controller" enabled="true">
+            <boolProp name="TransactionController.parent">false</boolProp>
+          </TransactionController>
+          <hashTree>
+            <WhileController guiclass="WhileControllerGui" testclass="WhileController" testname="While Controller" enabled="true">
```

Comment:
Good point. Calls to the backend are not asynchronous. To confirm, I spoke with Zak Burke. Following is the response:

> So-called bulk renewal happens by iterating through a list of loans (in `renew`) and making POST requests to renew-by-barcode one-by-one (in `renewItem`) until the list is exhausted. `renewItem` returns a promise, and `renew` calls  `await renewItem(...)` to handle the list iteratively so as not to overwhelm Okapi with too many simultaneous requests.

Reply:
I think waiting for a promise behaves differently than iteratively calling the API one by one in a loop.  In the former case many requests can be sent near concurrently because the response of each request is not needed until the end. In the latter case calling the API one by one means that once an API call is made the response has to come back before the next request is issued.  Is this also your understanding of "promise" code?

Reply:
They are using `await` which does not make it concurrent. It is still iterative. Here is the link to [code](https://github.com/folio-org/ui-users/blob/9c413c399a1a75589da1b44ad350f412ffd12ec8/src/components/Wrappers/withRenew.js#L130-L174).
From what I know, `promise` will be either `resolve` or `reject` within a callback. And a callback may or may not be async.

Reply:
Very good findings.  Thanks for digging it further.